### PR TITLE
[DISCO-2785] Update accuweather provider name and remove accuweather_city

### PR DESCRIPTION
**Pull request by tiftran**
## References

JIRA: [DISCO-2785](https://mozilla-hub.atlassian.net/browse/DISCO-2785)

## Description

Ripped out the old accuweather provider, renamed anything "accuweather_city" to "accuweather"

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [ ] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2785]: https://mozilla-hub.atlassian.net/browse/DISCO-2785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/unit/providers/weather/backends/test_accuweather.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
 from merino.providers.weather.backends.protocol import (
```
Comment (tiftran) [inline tests/unit/providers/weather/backends/test_accuweather.py:33]: This was previously `test_accuweather_city.py` I renamed it and deleted `test_accuweather.py`

File: merino/providers/weather/backends/accuweather.py [hunk 2]
```diff
@@ -954,7 +459,7 @@ async def get_weather_report_with_location_key(
         """
         # Look up for all the weather data from the cache.
         try:
-            with self.metrics_client.timeit("accuweather.cache.fetch-via-location-key"):
+            with self.metrics_client.timeit("accuweather.cache.fetch"):
```
Comment (ncloudioj) [inline merino/providers/weather/backends/accuweather.py:462]: Maybe we should keep `-via-location-key` to differentiate this from fetch the other cache fetch as the cache keys are different.

File: merino/providers/weather/backends/accuweather.py [hunk 3]
```diff
@@ -973,9 +478,7 @@ async def get_weather_report_with_location_key(
                 )
         except CacheAdapterError as exc:
             logger.error(f"Failed to fetch weather report from Redis: {exc}")
-            self.metrics_client.increment(
-                "accuweather.cache.fetch-via-location-key.error"
-            )
+            self.metrics_client.increment("accuweather.cache.fetch.error")
```
Comment (ncloudioj) [inline merino/providers/weather/backends/accuweather.py:481]: Ditto.

**Review by ncloudioj — APPROVED** [review]
Looks great, thanks!

**Pull request merged**